subtests/docker_cli/events/events.py

```diff
@@ -34,58 +36,13 @@
 from dockertest.xceptions import DockerValueError
 
 
-cid_regex = re.compile(r'\s+([a-z0-9]{64})\:\s+')
-fqin_regex = DockerImage.repo_split_p
-source_regex = re.compile(r'\s+\(from\s+\S+\)\s+')
-operation_regex = re.compile(r'\s+(\w+)$')  # final word chars
-
-
-def event_dt(line):
-    try:
-        return DockerTime(line)
-    except (AttributeError, TypeError, ValueError):  # failed
-        return None
-
-
-def event_cid(line):
-    mobj = cid_regex.search(line)
-    if mobj is not None:
-        return mobj.group(1)
-    else:
-        return None
-
-
-def event_fqin(line):
-    mobj = fqin_regex.search(line)
-    if mobj is not None:
-        return mobj.group(0)
-    else:
-        return None
-
-
-def event_source(line):
-    mobj = source_regex.search(line)
-    if mobj is not None:
-        # Verifies formatting
-        containerid = "".join([s for s in mobj.group(0)
-                               if s is not None])
-        return containerid
-    else:
-        return None
-
-
-def event_operation(line):
-    mobj = operation_regex.search(line)
-    if mobj is not None:
-        return mobj.group(1)
-    else:
-        return None
-
-
-def event_details(line):
-    return {'datetime': event_dt(line),
-            'source': event_source(line),
-            'operation': event_operation(line)}
+regexes = {
+    'timestamp': r'[\d-]+T[\d:]+\.\d+[+-][\d:]+',   # <iso8601>.<µs>[+/-]HH:MM
+    'cid':       r'(sha256:)?[0-9a-fA-F]{64}',      # 64-char hash
+    'fqin':      DockerImage.repo_split_p,          # eg some.repo/image:tag
```

Comment:
Okay, I C, ya, just use ``DockerImage.repo_split_p.patern``  that will "un-compile" it for you :smile: